### PR TITLE
[6X backport] use TRUSTED_SHELL in gpinitsystem

### PR DESCRIPTION
**Pull request by sruthicp**
**Issue**: 
When a segment is created using gpcreateseg.sh, trusted-shell from the configuration file is not used.

**Cause**: 
gpinitsystem when creating a segment by calling gpcreateseg.sh script, environment variable TRUSTED_SHELL is not passed. This resulted in using the default shell every time during segment creation.

**Solution**: 
When calling gpcreateseg.sh, set the environment variable to the correct value. This makes sure that TRUSTED_SHELL is set in the gpcreateseg.sh script.

**Testing Done**:
1. Created a demo cluster locally
2. Ran concourse pipeline and made sure its green
3. Manual testing
   - Updated gpcreatesegment.sh file to print TRUSTED_SHELL then ran make create-demo-cluster, Which prints the TRUSTED_SHELL value as "~/workspace/gpdb/gpAux/gpdemo/lalshell"
   - Updated the value of TRUSTED_SHELL in clusterConfigFile to "ssh" and ran make create-demo-cluster to check if the updated value is correctly passing to gpcreateseg.sh from gpinitsystem.
4. Added behave test cases for the change

 Parent PR: https://github.com/greenplum-db/gpdb/pull/15269. 
7x PR for behave test - https://github.com/greenplum-db/gpdb/pull/15382

### CODE REVIEW

**Comment by shrakesh** [comment]
@sruthicp In my opinion we should automate the manual test which can be put in behave test case. a single scenario covering both cases would be good to have.

File: gpMgmt/bin/lib/gpcreateseg.sh [hunk 1]
```diff
@@ -20,7 +20,7 @@
 # Prep Code
 
 WORKDIR=`dirname $0`
-
+echo "Trusted shell is:" $TRUSTED_SHELL
```
Comment (shrakesh) [inline gpMgmt/bin/lib/gpcreateseg.sh:23]: do we require this statement?

**Comment by piyushc01** [comment]
Update coming message to contain issue, cause, solution and testing done. 
Also update testing done section, missed to mention newly added functional tests. 
Make sure when merging, have both changes squashed in single commit.

File: gpMgmt/test/behave/mgmt_utils/gpinitsystem.feature [hunk 2]
```diff
@@ -349,3 +349,23 @@ Feature: gpinitsystem tests
          #restore hosts file
          And restore /etc/hosts file and cleanup hostlist file
 
+    Scenario: gpinitsystem should pass the default value of trusted_shell properly to gpcreateseg
+        Given create demo cluster config
+        And the user runs command "rm -rf /tmp/gpinitsystemtest && mkdir /tmp/gpinitsystemtest"
+        When the user runs command "gpinitsystem -a -c ../gpAux/gpdemo/clusterConfigFile -l /tmp/gpinitsystemtest -s localhost -P 21100 -S /wrong/path -h ../gpAux/gpdemo/hostfile --ignore-warnings"
```
Comment (shrakesh) [inline gpMgmt/test/behave/mgmt_utils/gpinitsystem.feature:355]: do we require standby to be configured for this scenario? Check if these steps can be optimized.

Comment (sruthicp) [reply gpMgmt/test/behave/mgmt_utils/gpinitsystem.feature]: Removed standby configuration

File: gpMgmt/test/behave/mgmt_utils/gpinitsystem.feature [hunk 3]
```diff
@@ -349,3 +349,23 @@ Feature: gpinitsystem tests
          #restore hosts file
          And restore /etc/hosts file and cleanup hostlist file
 
+    Scenario: gpinitsystem should pass the default value of trusted_shell properly to gpcreateseg
+        Given create demo cluster config
+        And the user runs command "rm -rf /tmp/gpinitsystemtest && mkdir /tmp/gpinitsystemtest"
```
[the remark this answers is not in the document]
Comment (sruthicp) [reply gpMgmt/test/behave/mgmt_utils/gpinitsystem.feature:354]: Done

**Review by shrakesh — APPROVED** [review]
Approved with minor comment. 
looking good to me.

File: gpMgmt/test/behave/mgmt_utils/gpinitsystem.feature [hunk 4]
```diff
@@ -349,3 +349,23 @@ Feature: gpinitsystem tests
...
+
+    Scenario: gpinitsystem should pass the changed value of trusted_shell properly to gpcreateseg
+        Given create demo cluster config
+        And the user runs command "sed -i.bak -E 's/TRUSTED_SHELL=.*/TRUSTED_SHELL=ssh/g' ../gpAux/gpdemo/clusterConfigFile"
+        And the user runs command "rm -rf /tmp/gpinitsystemtest && mkdir /tmp/gpinitsystemtest"
+        When the user runs command "gpinitsystem -a -c ../gpAux/gpdemo/clusterConfigFile -l /tmp/gpinitsystemtest -h ../gpAux/gpdemo/hostfile --ignore-warnings"
+        Then gpinitsystem should return a return code of 0
+        When the user runs command "grep -q '.*gpcreateseg\.sh.*Completed ssh.*' /tmp/gpinitsystemtest/gpinitsystem_*.log; echo $?"
+        Then grep should print "0" to stdout
+        And the user runs command "rm ../gpAux/gpdemo/clusterConfigFile && mv ../gpAux/gpdemo/clusterConfigFile.bak ../gpAux/gpdemo/clusterConfigFile"
```
[the remark this answers is not in the document]
Comment (sruthicp) [reply gpMgmt/test/behave/mgmt_utils/gpinitsystem.feature]: yes, It is not required. Removed it

File: gpMgmt/test/behave/mgmt_utils/gpinitsystem.feature [hunk 5]
```diff
@@ -349,3 +349,23 @@ Feature: gpinitsystem tests
          #restore hosts file
          And restore /etc/hosts file and cleanup hostlist file
 
+    Scenario: gpinitsystem should pass the default value of trusted_shell properly to gpcreateseg
+        Given create demo cluster config
+        And the user runs command "rm -rf /tmp/gpinitsystemtest && mkdir /tmp/gpinitsystemtest"
+        When the user runs command "gpinitsystem -a -c ../gpAux/gpdemo/clusterConfigFile -l /tmp/gpinitsystemtest -h ../gpAux/gpdemo/hostfile --ignore-warnings"
+        Then gpinitsystem should return a return code of 0
+        When the user runs command "grep -q '.*gpcreateseg\.sh.*lalshell.*' /tmp/gpinitsystemtest/gpinitsystem_*.log; echo $?"
```
Comment (piyushc01) [inline gpMgmt/test/behave/mgmt_utils/gpinitsystem.feature:357]: Can we check explicit log file?

File: gpMgmt/test/behave/mgmt_utils/gpinitsystem.feature [hunk 6]
```diff
@@ -349,3 +349,23 @@ Feature: gpinitsystem tests
...
+        When the user runs command "gpinitsystem -a -c ../gpAux/gpdemo/clusterConfigFile -l /tmp/gpinitsystemtest -h ../gpAux/gpdemo/hostfile --ignore-warnings"
+        Then gpinitsystem should return a return code of 0
+        When the user runs command "grep -q '.*gpcreateseg\.sh.*Completed .*lalshell.*' /tmp/gpinitsystemtest/gpinitsystem_*.log"
+        Then grep should return a return code of 0
+        And the user runs command "rm -rf /tmp/gpinitsystemtest"
+
+    Scenario: gpinitsystem should pass the changed value of trusted_shell properly to gpcreateseg
+        Given create demo cluster config
+        And the user runs command "sed -i.bak -E 's/TRUSTED_SHELL=.*/TRUSTED_SHELL=ssh/g' ../gpAux/gpdemo/clusterConfigFile"
+        And the user runs command "rm -rf /tmp/gpinitsystemtest && mkdir /tmp/gpinitsystemtest"
```
Comment (piyushc01) [inline gpMgmt/test/behave/mgmt_utils/gpinitsystem.feature:364]: Need this step anymore as with every test we will be doing the cleanup?

File: gpMgmt/test/behave/mgmt_utils/gpinitsystem.feature (hunk 5 above)
Comment (sruthicp) [reply gpMgmt/test/behave/mgmt_utils/gpinitsystem.feature]: Updated

File: gpMgmt/test/behave/mgmt_utils/gpinitsystem.feature (hunk 6 above)
Comment (sruthicp) [reply gpMgmt/test/behave/mgmt_utils/gpinitsystem.feature]: Removed it

File: gpMgmt/test/behave/mgmt_utils/gpinitsystem.feature (hunk 3 above)
Comment (piyushc01) [inline gpMgmt/test/behave/mgmt_utils/gpinitsystem.feature:354]: No need of this line as every test is taking care of cleanup

**Review by piyushc01 — APPROVED** [review]
Looks good to me.

**Review by jnihal — APPROVED** [review]

**Pull request closed without merging**